autoPyTorch/pipeline/components/training/trainer/GridCutMixTrainer.py

```diff
@@ -0,0 +1,68 @@
+import typing
+
+import numpy as np
+
+import torch
+
+from autoPyTorch.pipeline.components.training.trainer.base_trainer import BaseTrainerComponent
+from autoPyTorch.pipeline.components.training.trainer.mixup_utils import MixUp
+
+
+class GridCutMixTrainer(MixUp, BaseTrainerComponent):
+
+    def data_preparation(self, X: np.ndarray, y: np.ndarray,
+                         ) -> typing.Tuple[np.ndarray, typing.Dict[str, np.ndarray]]:
+        """
+        Depending on the trainer choice, data fed to the network might be pre-processed
+        on a different way. That is, in standard training we provide the data to the
+        network as we receive it to the loader. Some regularization techniques, like mixup
+        alter the data.
+
+        Args:
+            X (np.ndarray): The batch training features
+            y (np.ndarray): The batch training labels
+
+        Returns:
+            np.ndarray: that processes data
+            typing.Dict[str, np.ndarray]: arguments to the criterion function
+        """
+        beta = 1.0
+        lam = self.random_state.beta(beta, beta)
+        batch_size, channel, W, H = X.size()
```

Comment:
If `X` is really `np.ndarray`,
```suggestion
        batch_size, _, W, H = X.shape
```
In any case, `.shape` works for both `np.ndarray` and `torch.Tensor`, so I prefer `.shape`.